search/include/pcl/search/organized.h

```diff
@@ -221,15 +221,20 @@ namespace pcl
             float dist_y = point.y - query.y;
             float dist_z = point.z - query.z;
             float squared_distance = dist_x * dist_x + dist_y * dist_y + dist_z * dist_z;
-            if (queue.size () < k)
+            const auto queue_size = queue.size ();
+            if (queue_size < k)
             {
-              queue.push (Entry (index, squared_distance));
-              return queue.size () == k;
+              queue.emplace (std::upper_bound (queue.begin(), queue.end(), squared_distance,
```

Comment:
Perhaps, extract the emplace into a lambda

Reply:
Done, now it is even a few percents faster, though I don't really see why

Reply:
We should really have some benchmarks built in the CI somehow

Reply:
Yes, that would be great. I see you already opened an issue for that, though there does not seem to be much progress lately.